### PR TITLE
feat(onboarding): ask signup referral source

### LINKED ISSUE #7873: Add a 'where did you first hear of us' text area to sign up flow
## Is your feature request related to a problem?

We don't have a great grasp of where our users _actually_ first hear about us.

As much as we can try to attribute through UTM tags, the reality is people will have read an article about us on HackerNews and then later googled us.

If we can understand this better, we can focus our spend on dev rel and marketing properly.

## Describe the solution you'd like

A mandatory text area in the sign up flow when the organization is created (for cloud) that asks where users first _heard_ of us. It should be a free form text field so we get more open answers.

## Describe alternatives you've considered

Spend marketing money in the wrong places, potentially.

### PR DESCRIPTION
**Pull request by kappa90**
## Problem

We do not track from where users heard about PostHog during signup.
Closes #7873

Considered adding this as an experiment/feature flag, but I believe leaving it as an optional field is enough and doesn't block users from signing up

## Changes

- Add a new input to the signup form
- a new field `referral_source` is now added to the event `user signed up`

Screenshot 2022-08-31 at 19 58 59

## How did you test this code?
- Ran through the signup manually, and checked that the new field is sent to the analytics

### CODE REVIEW

File: ee/clickhouse/queries/test/__snapshots__/test_trends.ambr [hunk 1]
```diff
@@ -332,7 +332,7 @@
        AND timestamp <= toDateTime('2020-01-02 23:59:59')
        AND (has(['technology'], replaceRegexpAll(JSONExtractRaw(group0_properties, 'industry'), '^"|"$', ''))) )
   GROUP BY actor_id
-  LIMIT 200
+  LIMIT 100
```
Comment (benjackwhite) [inline ee/clickhouse/queries/test/__snapshots__/test_trends.ambr:335]: Why is this here...? Feels like the diff I was seeing from one of my unmerged PRs...

**Review by benjackwhite — CHANGES_REQUESTED** [review]
I think your `black` config might  be off as there are a lot of formatting changes that shouldn't be here. 

Also a snapshot file that almost seems like its from another PR?

File: ee/clickhouse/queries/test/__snapshots__/test_trends.ambr (hunk 1 above)
Comment (kappa90) [reply ee/clickhouse/queries/test/__snapshots__/test_trends.ambr:335]: I didn't do that `Update snapshots` commit? wtf?

Comment (kappa90) [reply ee/clickhouse/queries/test/__snapshots__/test_trends.ambr:335]: I had to nuke my laptop a few days ago, and reinstall everything, checking what happened with `black` 🤦

Comment (neilkakkar) [reply ee/clickhouse/queries/test/__snapshots__/test_trends.ambr]: Sorry about the snapshots, my PR went in, didn't update this, but then master started failing, so a subsequent PR fixed this. You were in a race condition as you started your PR after the first one went in, but before the second one did 😂

**Review by benjackwhite — APPROVED** [review]
